sdk/identity/Azure.Identity/src/CredentialDiagnosticScope.cs

```diff
@@ -79,7 +80,24 @@ private bool TryWrapException(ref Exception exception, string additionalMessageT
                     return true;
                 }
             }
-            string exceptionMessage = $"{_name.Substring(0, _name.IndexOf('.'))} authentication failed: {exception.Message}";
+
+            string exceptionMessage = $"{_name.Substring(0, _name.IndexOf('.'))} authentication failed: ";
+
+            if (exception is MsalServiceException mse)
+            {
+                if (mse.ErrorCode == "user_assigned_managed_identity_not_supported")
+                {
+                    exceptionMessage += Constants.MiSourceNoUserAssignedIdentityMessage;
+                }
+                else if (mse.ErrorCode == "managed_identity_request_failed")
```

Comment:
I don't know that we can guaruntee they are stable, but I believe that changing them would be a breaking change. I'll confirm with MSAL.

We currently only append for this case because the error message is generic enough. Otherwise, we want to control the primary exception message. In all cases the full MSAL exception will be in the inner exception, so the information is never lost.

Reply:
confirmed they are stable